react/FilterNavigator.js

```diff
@@ -66,6 +73,101 @@ class FilterNavigator extends Component {
     hiddenFacets: {},
   }
 
+  state = {
+    openContent: false,
+    filtersChecks: {},
+    pagesArgs: {},
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const nonEmptyFilters = this.mapCheckboxFilters(this.filters.filter(spec => spec.options.length > 0))
+
+    if (this.willUpdateFilters(Object.keys(nonEmptyFilters), Object.keys(prevState.filtersChecks))) {
+      this.setState({ filtersChecks: nonEmptyFilters })
+    }
+  }
+
+  willUpdateFilters = (newFilters, prevFilters) => {
+    if (newFilters.length !== prevFilters.length) return true
+    return !newFilters.every(e => prevFilters.includes(e))
+  }
+
+  mapCheckboxFilters = (filters) => {
+    const { map, rest } = this.props
+
+    const mapperOption = (isSelected, selectedFilters) => opt => {
+      const pagesArgs = formatFacetToLinkPropsParam(opt.type, opt)
+      return {
+        name: opt.Name,
+        ...pagesArgs,
+        checked: isSelected || !!find(propEq('Name', opt.Name), selectedFilters)
+      }
+    }
+    const mapperOptionsForFilter = (map, rest) => ({ options, type }) => mountOptions(options, type, map, rest).map(mapperOption(false, this.selectedFilters))
+    const reducerForOptions = (current, next) => union(current, next)
```

Comment:
Use `flatten` instead of the reducer, it will be cleaner

I think it can substitute the rest operator on line 109 also

https://ramdajs.com/docs/#flatten